src/__internal__/focus-trap/focus-trap-utils.js

```diff
@@ -1,12 +1,40 @@
 const defaultFocusableSelectors =
   'button:not([disabled]), [href], input:not([type="hidden"]):not([disabled]), select:not([disabled]), textarea:not([disabled]), [tabindex]';
 
+// callback-based rather than Promise-based, as Jest fake timers unfortunately don't work nicely with Promises
+const waitForCss = (element, expectedStyles, maxTime, callback) => {
+  const INTERVAL = 10;
+  let timeSoFar = 0;
+
+  const stylesMatch = () => {
+    const actualStyles = window.getComputedStyle(element);
+    return Object.entries(expectedStyles).every(
+      ([property, value]) => actualStyles[property] === value
+    );
+  };
+
+  const check = () => {
+    /* istanbul ignore else */
+    if (stylesMatch()) {
+      callback();
+    } else if (timeSoFar < maxTime) {
+      setTimeout(check, INTERVAL);
+      timeSoFar += INTERVAL;
+    }
+    // just "fail" silently if maxTime exceeded - callback will never be called
+  };
+
+  check();
+};
+
 function setElementFocus(element) {
   if (typeof element === "function") {
     element();
   } else {
     const el = element.current || element;
-    el.focus();
+    // sometimes (eg on MenuFullScreen) the .focus call happens before the wrapper element has CSS visibility set
+    // to visible - so we need to wait for the element to be visible otherwise the focusing fails silently
```

Comment:
I think this second comment can be removed as there's a comment in the above about it failing silently